### PR TITLE
[multistage] Register theta sketch aggregation functions in v2 query engine

### PR DESCRIPTION
**Pull request by xiangfu0**
Thanks to https://github.com/apache/pinot/pull/11105 , now we can support literal in aggregation functions.

- Register `distinctCountThetaSketch` and `distinctCountRawThetaSketch` in `AggregationFunctionType`
- Enable `ThetaSketchIntegrationTest` for v2 query engine.

### CODE REVIEW

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java [hunk 1]
```diff
@@ -88,8 +88,12 @@ public enum AggregationFunctionType {
   DISTINCTCOUNTRAWHLL("distinctCountRawHLL"),
   DISTINCTCOUNTSMARTHLL("distinctCountSmartHLL"),
   FASTHLL("fastHLL"),
-  DISTINCTCOUNTTHETASKETCH("distinctCountThetaSketch"),
-  DISTINCTCOUNTRAWTHETASKETCH("distinctCountRawThetaSketch"),
+  DISTINCTCOUNTTHETASKETCH("distinctCountThetaSketch", ImmutableList.of("DISTINCT_COUNT_THETA_SKETCH"),
+      SqlKind.OTHER_FUNCTION, SqlFunctionCategory.USER_DEFINED_FUNCTION, OperandTypes.VARIADIC, ReturnTypes.BIGINT,
```
Comment (walterddr) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java:92]: VARIADIC is problematic. we can only run these in v1. i don't know if we should do this and allow the problematic syntax in v2

**Review by walterddr — APPROVED** [review]

**Review by walterddr — COMMENTED** [review]
looks good to me in general. one question is similar to `HISTOGRAM` and `PERCENTILE`
see: 
https://github.com/apache/calcite/blob/bdfb17029f7e205f895dc3dfd0f37c8ff2520823/core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java#L2257
and 
https://github.com/apache/calcite/blob/bdfb17029f7e205f895dc3dfd0f37c8ff2520823/core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java#L1211

Basically
- HISTOGRAM split into HISTOGRAM_AGG method that return a histogram type, and the "extract final result method" from that histogram
- PERCENTILE is actually an attribute with the SqlBasicAggFunction class (similar to distinct attribute) but it is still returning a percentile type, then extract the percentile (as a standard function)

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java [hunk 2]
```diff
@@ -88,8 +88,14 @@ public enum AggregationFunctionType {
...
-  DISTINCTCOUNTTHETASKETCH("distinctCountThetaSketch"),
-  DISTINCTCOUNTRAWTHETASKETCH("distinctCountRawThetaSketch"),
+  DISTINCTCOUNTTHETASKETCH("distinctCountThetaSketch", ImmutableList.of("DISTINCT_COUNT_THETA_SKETCH"),
+      SqlKind.OTHER_FUNCTION, SqlFunctionCategory.USER_DEFINED_FUNCTION,
+      OperandTypes.family(ImmutableList.of(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER), ordinal -> ordinal > 0),
+      ReturnTypes.BIGINT, ReturnTypes.explicit(SqlTypeName.OTHER)),
+  DISTINCTCOUNTRAWTHETASKETCH("distinctCountRawThetaSketch", ImmutableList.of("DISTINCT_COUNT_RAW_THETA_SKETCH"),
+      SqlKind.OTHER_FUNCTION, SqlFunctionCategory.USER_DEFINED_FUNCTION,
+      OperandTypes.family(ImmutableList.of(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER), ordinal -> ordinal > 0),
+      ReturnTypes.BIGINT, ReturnTypes.explicit(SqlTypeName.OTHER)),
```
Comment (walterddr) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java:98]: RAW return type is a base64 encoded string
```suggestion
      ReturnTypes.VARCHAR, ReturnTypes.explicit(SqlTypeName.OTHER)),
```

**Review by walterddr — COMMENTED** [review]
also see: https://github.com/apache/pinot/pull/11153

**Review by walterddr — APPROVED** [review]

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java [hunk 3]
```diff
@@ -88,8 +88,14 @@ public enum AggregationFunctionType {
...
-  DISTINCTCOUNTTHETASKETCH("distinctCountThetaSketch"),
-  DISTINCTCOUNTRAWTHETASKETCH("distinctCountRawThetaSketch"),
+  DISTINCTCOUNTTHETASKETCH("distinctCountThetaSketch", ImmutableList.of("DISTINCT_COUNT_THETA_SKETCH"),
+      SqlKind.OTHER_FUNCTION, SqlFunctionCategory.USER_DEFINED_FUNCTION,
+      OperandTypes.family(ImmutableList.of(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER), ordinal -> ordinal > 0),
+      ReturnTypes.BIGINT, ReturnTypes.explicit(SqlTypeName.OTHER)),
+  DISTINCTCOUNTRAWTHETASKETCH("distinctCountRawThetaSketch", ImmutableList.of("DISTINCT_COUNT_RAW_THETA_SKETCH"),
+      SqlKind.OTHER_FUNCTION, SqlFunctionCategory.USER_DEFINED_FUNCTION,
+      OperandTypes.family(ImmutableList.of(SqlTypeFamily.ANY, SqlTypeFamily.CHARACTER), ordinal -> ordinal > 0),
+      ReturnTypes.VARCHAR_2000, ReturnTypes.explicit(SqlTypeName.OTHER)),
```
Comment (walterddr) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/AggregationFunctionType.java:98]: this ReturnType is suppose to be VARBINARY right?
